tests_e2e/pipeline/pipeline.yml

```diff
@@ -73,36 +62,48 @@ jobs:
   - job: "ExecuteTests"
 
     steps:
+      - task: UsePythonVersion@0
+        displayName: "Set Python Version"
+        inputs:
+          versionSpec: '3.10'
+          addToPath: true
+          architecture: 'x64'
+
+      # Extract the Azure cloud from the "connection_info" variable and store it in the "cloud" variable.
+      # The cloud name is used as a suffix of the value for "connection_info" and comes after the last '-'.
+      - bash: echo "##vso[task.setvariable variable=cloud]$(echo $CONNECTION_INFO | sed 's/^.*-//')"
+        displayName: "Set Cloud type"
+
       - task: DownloadSecureFile@1
         name: downloadSshKey
         displayName: "Download SSH key"
         inputs:
           secureFile: 'id_rsa'
 
       - task: AzureKeyVault@2
-        displayName: "Fetch secrets from KV"
+        displayName: "Fetch connection info"
         inputs:
-          azureSubscription: '$(azureConnection)'
-          KeyVaultName: 'dcrV2SPs'
+          azureSubscription: 'azuremanagement'
+          KeyVaultName: '$(connection_info)'
           SecretsFilter: '*'
-          RunAsPreJob: true
```

Comment:
'RunAsPreJob' makes the secrets available to all the tasks, this is not needed since the only task that uses them is "Execute tests". I removed this parameter to use the default value (false), which makes the secrets available only to the next task and I moved the task to precede "Execute tests".